tests/test_bounds.sh

```diff
@@ -1,21 +1,19 @@
 #!/bin/sh
-# This script is a draft. Note: Bash is not available on CheriBSD.
-# This is a simple way to test the examples by checking ``stderr``.
+# This is a simple way to test "bounds.c" by checking ``stderr``.
+set -e
 
 : "${SSHPORT:=10021}"
-RESULT_riscv64="fail"
+RESULT="fail"
 if [ -f "../bin/bounds" ]; then
     scp -P $SSHPORT ../bin/bounds root@127.0.0.1:/root
-	RESULT_riscv64=`ssh -p $SSHPORT root@127.0.0.1 -t 'sh -c "/root/bounds 2>&1 > /dev/null"'`
-    echo $RESULT_riscv64
-    # note: other forms of equality (test) fail on SH
-    if [ -z "${RESULT_riscv64##*In-address space security exception (core dumped)*}" ] ;then
-        echo "TEST PASSED!"
+    RESULT={$(ssh -p $SSHPORT -t root@127.0.0.1 'sh -c "/root/bounds 2>&1 > /dev/null"') && exit 1} || true
```

Comment:
There is one confusing thing here: if `bounds` fails, nothing will be printed out. I wonder if we want to do something like this:

```sh
echo -n "bounds... "
<run ssh command>
exit_status=0
{{RESULT=$(ssh -p $SSHPORT -t root@127.0.0.1 'sh -c "/root/bounds 2>&1 > /dev/null"')} && exit_status=1} || true
if [ $exit_status != 0 ]; then
  echo "failed"
  echo $RESULT
else
  echo "passed"
fi
```

(Warning: not tested in any way!)

Reply:
Alright.

Reply:
Hang on, why do you see that "PASSED" there? I've followed the cargo suggestion with "ok".
Maybe, comments (previous) addressed in 020d11d, was necessary before writing "Ready for review"? My bad, sorry.

Reply:
The `exit 1` means that if the command exited with a 0 exit code we don't reach the `echo`s on the following lines.

Reply:
> The `exit 1` means that if the command exited with a 0 exit code we don't reach the `echo`s on the following lines.

Yes, my question was related to the code shown not to your comment. The code in the PR, not your code.

Reply:
Anyway, I'll address and test and also I want to let you know that I have replaced "PASSED" with "OK", so it is less chatty, as you like :) More words in case it fails, instead.

Reply:
> Yes, my question was related to the code shown not to your comment. The code in the PR, not your code.

Yes, I was referring to the code in the PR.

Reply:
> > Yes, my question was related to the code shown not to your comment. The code in the PR, not your code.
> 
> Yes, I was referring to the code in the PR.

But the code in the PR is [this](https://github.com/capablevms/cheri-examples/commit/020d11d9ec60a137b3a1eb70a700cb79526783e7#diff-dee479da72cd420f474a565af7d5db58e9ea75697240cccf3445d0669e7428d0). Again, your comment is correct but the code you commented is outdated. This won't change the meaning of what you said. I am just trying to understand why it is outdated.

Reply:
I promise I am referring to https://github.com/capablevms/cheri-examples/commit/020d11d9ec60a137b3a1eb70a700cb79526783e7#diff-dee479da72cd420f474a565af7d5db58e9ea75697240cccf3445d0669e7428d0 :) Notice that the `exit 1` happens *before* the `echo "bounds... FAILED!`. In other words, if it fails, the user won't be told that.